soc/nordic/nrf54h/gpd/gpd.c

```diff
@@ -211,6 +212,33 @@ int nrf_gpd_release(uint8_t id)
 	return onoff_release(&gpd_mgr->mgr);
 }
 
+int nrf_gpd_retain_pins_set(const struct pinctrl_dev_config *pcfg, bool retain)
```

Comment:
Not sure why this function is in this unit. The function is about configuring pin properties, while the file is about global power domains. How about moving the function to pinctrl?